apps/cnquery/cmd/builder/parse.go

```diff
@@ -612,14 +635,38 @@ func ParseTargetAsset(cmd *cobra.Command, args []string, providerType providers.
 		}
 	case providers.ProviderType_GOOGLE_WORKSPACE:
 		connection.Backend = providerType
+
+		// env vars have precedence over the --credentials-path arg
+		credsPath := readEnvs("GOOGLEWORKSPACE_CREDENTIALS", "GOOGLEWORKSPACE_CLOUD_KEYFILE_JSON", "GOOGLE_CREDENTIALS")
+		if credsPath != "" {
```

Comment:
this seems to be identical for google workspace and gcp. Why not extract it in a function?

Reply:
Maybe not the full part but we could restructure it in somehthing like:

```go
var credsPaths []string
credsPaths = readEnvs("GOOGLEWORKSPACE_CREDENTIALS", "GOOGLEWORKSPACE_CLOUD_KEYFILE_JSON", "GOOGLE_CREDENTIALS"))

if credsPath, err := cmd.Flags().GetString("credentials-path"); err != nil {
    log.Fatal().Err(err).Msg("cannot parse --credentials-path value")
} else if credsPath != "" {
  credsPaths = append(credsPaths, credPath)
}

for i := range credsPaths {
   credPath := credsPaths[i]
   
  serviceAccount, err := os.ReadFile(credsPath)
  if err == nil {
    connection.Credentials = append(connection.Credentials, &vault.Credential{
	Type:   vault.CredentialType_json,
	Secret: serviceAccount,
     })
     break // abort loop since we found credential
  }
}
```